File: fabric/src/main/kotlin/cache/StreamListener.kt
package com.obecto.perper.fabric.cache

class StreamListener(
    var callerAgent: String,
    var callerInstance: String,
    var caller: String,
    var parameter: Int,
    var filter: HashMap<String, Any?>?,
    var replay: Boolean,
    var localToData: Boolean,
)

File: agent/dotnet/src/Perper/Protocol/Cache/Instance/StreamListener.cs
using System.Collections;

namespace Perper.Protocol.Cache.Instance
{
    public class StreamListener
    {
        public StreamListener(
            string callerAgent,
            string callerInstance,
            string caller,
            int parameter,
            bool replay,
            bool localToData,
            Hashtable? filter = null)
        {
            CallerAgent = callerAgent;
            CallerInstance = callerInstance;
            Caller = caller;
            Parameter = parameter;
            Filter = filter;
            Replay = replay;
            LocalToData = localToData;
        }

        public string CallerAgent { get; }

        public string CallerInstance { get; }

        public string Caller { get; }

        public int Parameter { get; }

        public Hashtable? Filter { get; }

        public bool Replay { get; }

        public bool LocalToData { get; }
    }
}